### ISSUE TITLE
Different Page Behaviour When Page Visited in a Cypress-Launched Browser versus Browser Launched Normally

### ISSUE THREAD

**Issue by todd-m-kemp** [cypress-io/cypress#7848]
### Current behaviour:

When visiting the page `https://share.vidyard.com/watch/nHEcBjJ9sHdD1mgfezkTmf` in Chrome launched via Cypress and executing the statement `window.vidyardEmbed.api.getPlayersByUUID('nHEcBjJ9sHdD1mgfezkTmf')[0].ready()` in the browser console, the output value is `false`.

### Desired behavior:

The output value should be `true`. This is what happens if you navigate to the page and execute the same statement in the console in Chrome launched normally (outside a Cypress test). The behaviour inside Cypress should mirror the behaviour outside of Cypress.

### Test code to reproduce

```
context('Cypress Issue', function() {

  beforeEach(function () {
    cy.visit('https://share.vidyard.com/watch/nHEcBjJ9sHdD1mgfezkTmf?')
    cy.window().then((window) => {
      expect(window.vidyardEmbed.api.getPlayersByUUID('nHEcBjJ9sHdD1mgfezkTmf')[0]._ready).to.equal(true)
    })
  })

  it('Test', function () {

  })
})
```

### Versions

Cypress Version: 4.4.1
Chrome Version: 83.0.4103.116
OS Version: macOS 10.14.6

**Comment by testn** [comment]
try to use should() instead of then()

**Comment by todd-m-kemp** [comment]
Thanks @testn; it doesn't look like that was your problem but your commented prompted my realization that the code I supplied to reproduce wasn't the greatest.  I've updated the original report.

**Comment by jennifer-shehane** [comment]
Yeah, I can recreate this. I'm not sure why this value is different inside of Cypress. I added a long `cy.wait()` to ensure it's not grabbing some static window object initially, but it still evaluates to `false`. 

```js
it('Test', () => {
  cy.visit('https://share.vidyard.com/watch/nHEcBjJ9sHdD1mgfezkTmf?')
  cy.window()
  .its('vidyardEmbed.api')
  .invoke('getPlayersByUUID', 'nHEcBjJ9sHdD1mgfezkTmf')
  .its('0._ready')
  .should('be.true') // fails
})

it('Test', () => {
  cy.visit('https://share.vidyard.com/watch/nHEcBjJ9sHdD1mgfezkTmf?')
  cy.window().then((win) => {
    // fails
    expect(win.vidyardEmbed.api.getPlayersByUUID('nHEcBjJ9sHdD1mgfezkTmf')[0]._ready).to.be.true
  })
})
```

**Comment by todd-m-kemp** [comment]
This issue is still reproducible in Cypress 12.8.1. in Chrome 113 on macOS 13.3.1.

**Comment by chrisbreiding** [comment]
Thanks for confirming, @todd-m-kemp!

**Comment by chrisbreiding** [comment]
I also replicated this issue in the latest version of Cypress and will forward this ticket to the appropriate team. They will evaluate the priority of this ticket and consider their capacity to pick it up. Please note that this does not guarantee that this issue will be resolved.
